sys/include/net/ng_pktbuf.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    ng_pktbuf   Packet buffer
+ * @ingroup     net
+ * @brief       A global network packet buffer.
+ *
+ * @note    **WARNING!!** Do not store data structures that are not packed
+ *          (defined with `__attribute__((packed))`) or enforce alignment in
+ *          in any way in here if @ref NG_PKTBUF_SIZE > 0. On some RISC architectures
+ *          this *will* lead to alignment problems and can potentially result
+ *          in segmentation/hard faults and other unexpected behaviour.
+ *
+ * @{
+ *
+ * @file
+ * @brief   Interface definition for the global network buffer. Network devices
+ *          and layers can allocate space for packets here.
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef NG_PKTBUF_H_
+#define NG_PKTBUF_H_
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "atomic.h"
+#include "cpu-conf.h"
+#include "net/ng_pkt.h"
+#include "net/ng_nettype.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @def     NG_PKTBUF_SIZE
+ * @brief   Maximum size of the static packet buffer.
+ *
+ * @details The rational here is to have at least space for 4 full-MTU IPv6
+ *          packages (2 incoming, 2 outgoing; 2 * 2 * 1280 B = 5 KiB) +
+ *          Meta-Data (roughly estimated to 1 KiB; might be smaller). If
+ *          @ref NG_PKTBUF_SIZE is 0 the packet buffer will use dynamic memory
+ *          management to allocate packets.
+ */
+#ifndef NG_PKTBUF_SIZE
+#define NG_PKTBUF_SIZE  (6144)
+#endif  /* NG_PKTBUF_SIZE */
+
+#if defined(DEVELHELP) && (NG_PKTBUF_SIZE > 0)
+/**
+ * @brief   All-over maximum number of bytes allocated in packet buffer during
+ *          runtime
+ *
+ * @detail  This variable is only available if @ref DEVELHELP is active and
+ *          @ref NG_PKTBUF_SIZE > 0.
+ */
+extern unsigned int ng_pktbuf_max_bytes;
```

Comment:
Do you have any special use case in mind to make this variable global? I think a function like `ng_pktbuf_stats()` that is just available if compiled with `DEVELHELP` and that just prints this kind of data should be sufficient.